src/jitlog/generator.lua

```diff
@@ -0,0 +1,780 @@
+local format = string.format
+local tinsert = table.insert
+
+local builtin_types = {
+  char = {size = 1, signed = true, printf = "%i", c = "char",  argtype = "char"},
+  
+  i8  = {size = 1, signed = true,  printf = "%i",   c = "int8_t",   argtype = "int32_t"},
+  u8  = {size = 1, signed = false, printf = "%i",   c = "uint8_t",  argtype = "uint32_t"},
+  i16 = {size = 2, signed = true,  printf = "%i",   c = "int16_t",  argtype = "int32_t"},
+  u16 = {size = 2, signed = false, printf = "%i",   c = "uint16_t", argtype = "uint32_t"},
+  i32 = {size = 4, signed = true,  printf = "%i",   c = "int32_t",  argtype = "int32_t"},
+  u32 = {size = 4, signed = false, printf = "%u",   c = "uint32_t", argtype = "uint32_t"},
+  i64 = {size = 8, signed = true,  printf = "%lli", c = "int64_t",  argtype = "int64_t"},
+  u64 = {size = 8, signed = false, printf = "%llu", c = "uint64_t", argtype = "uint64_t"},
+  
+  MSize  = {size = 4, signed = false,  printf = "%u", c = "uint32_t", argtype = "MSize"},
+  GCSize = {size = 4, signed = false,  printf = "%u", c = "GCSize", argtype = "GCSize"},
+
+  timestamp  = {size = 8, signed = false,  printf = "%llu", c = "uint64_t", writer = "timestamp_highres", noarg = true},
+  smallticks = {size = 4, signed = false,  printf = "%u",   c = "uint32_t", argtype = "uint64_t"},
+
+  GCRef      = {size = 4, signed = false, c = "GCRef", printf = "0x%llx", writer = "setref", ref = "gcptr32", ref64 = "gcptr64", argtype = "GCRef"},
+  --GCRef field with the value passed in as a pointer
+  GCRefPtr   = {size = 4, signed = false, c = "GCRef", printf = "0x%llx", writer = "setref", ref = "gcptr32", ref64 = "gcptr64", ptrarg = true, argtype = "void *"},
+  MRef       = {size = 4, signed = false, c = "MRef", printf = "0x%llx",  writer = "setref", ref = "ptr32", ref64 = "ptr64", ptrarg = true, argtype = "void *"},
+  -- Always gets widen to 64 bit since this is assumed not to be a gc pointer
+  ptr        = {size = 8, signed = false, c = "uint64_t", printf = "0x%llx", writer = "widenptr", ptrarg = true, argtype = "void *"},
+
+  string     = {vsize = true, printf = "%s", string = true,     c = "const char*", argtype = "const char *",  element_type = "char", element_size = 1},
+  stringlist = {vsize = true, printf = "%s", stringlist = true, c = "const char*", argtype = "const char *",  element_type = "char", element_size = 1},
+}
+
+local aliases = {
+  int8_t  = "i8",  uint8_t  = "u8",
+  int16_t = "i16", uint16_t = "u16",
+  int32_t = "i32", uint32_t = "u32",
+  int64_t = "i64", uint64_t = "u64",
+
+  bitfield = "u32",
+  bool = "u32", 
+}
+
+for name, def in pairs(aliases) do
+  assert(builtin_types[def])
+  builtin_types[name] = builtin_types[def]
+end
+
+if GC64 then
+  builtin_types.MSize = "uint64_t"
```

Comment:
does this still work with the types/aliases split?  maybe this `if` clause should go before the alias resolving loop, and set `aliases.MSize='uint64_t'`, instead of `builtin_types`

Reply:
It looks like msize is always 32 bits even in gc64 so the code is just wrong